File: backend/app/models/product.py
from .. import db

class Product(db.Model):
    __tablename__ = 'products'
    
    id = db.Column(db.Integer, primary_key=True)
    name = db.Column(db.String(100), nullable=False)
    description = db.Column(db.Text)
    category = db.Column(db.String(50), nullable=False)
    brand = db.Column(db.String(50), nullable=False)
    model = db.Column(db.String(50))
    price = db.Column(db.Float, nullable=False)
    image_url = db.Column(db.String(255))
    created_at = db.Column(db.DateTime, server_default=db.func.now())
    updated_at = db.Column(db.DateTime, server_default=db.func.now(), onupdate=db.func.now())
    
    # Relationships
    stocks = db.relationship('Stock', backref='product', lazy=True)
    order_details = db.relationship('OrderDetail', backref='product', lazy=True)
    reseller_stocks = db.relationship('ResellerStock', backref='product', lazy=True)
    return_requests = db.relationship('ReturnRequest', backref='product', lazy=True)
    
    def __repr__(self):
        return f'<Product {self.name}>'
    
    def to_dict(self):
        return {
            'id': self.id,
            'name': self.name,
            'description': self.description,
            'category': self.category,
            'brand': self.brand,
            'model': self.model,
            'price': self.price,
            'image_url': self.image_url,
            'created_at': self.created_at.isoformat() if self.created_at else None,
            'updated_at': self.updated_at.isoformat() if self.updated_at else None
        }

File: frontend/mobile/lib/models/product.dart
class Product {
  final int id;
  final String name;
  final String brand;
  final String category;
  final String model;
  final String description;
  final String imageUrl;
  final double price;

  Product({
    required this.id,
    required this.name,
    required this.brand,
    required this.category,
    required this.model,
    required this.description,
    required this.imageUrl,
    required this.price,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'],
      name: json['name'],
      brand: json['brand'],
      category: json['category'],
      model: json['model'],
      description: json['description'],
      imageUrl: json['image_url'],
      price: json['price'].toDouble(),
    );
  }
}

class ProductWithStock {
  final Product product;
  final int stock;

  ProductWithStock({required this.product, required this.stock});

  factory ProductWithStock.fromJson(Map<String, dynamic> json) {
    return ProductWithStock(
      product: Product.fromJson(json['product']),
      stock: json['stock'],
    );
  }
}
